pkg/reconciliation/reconcile_racks.go

```diff
@@ -2243,6 +2246,52 @@ func (rc *ReconciliationContext) fixMissingPVC() (bool, error) {
 	return false, nil
 }
 
+func (rc *ReconciliationContext) SetFullQueryLogging() result.ReconcileResult {
+	datacenter := rc.GetDatacenter()
+	var DCConfig map[string]interface{}
+	if err := json.Unmarshal(datacenter.Spec.Config, &DCConfig); err != nil {
+		rc.ReqLogger.Error(err, "error unmarshalling DC config JSON")
+		return result.RequeueSoon(2)
+	}
+	shouldFQLBeEnabled := false
+	if err := DCConfig["full_query_logging_options"]; err == nil {
+		serverMajorVersion, err := strconv.ParseInt(strings.Split(datacenter.Spec.ServerVersion, ".")[0], 10, 8)
```

Comment:
And a simple note.. serverVersion isn't necessarily correct (even if it should be)  - so careful here. User can set serverVersion to 3.11.11 and serverImage to 4.0. Or the other way around.

Reply:
That's a good point. In this instance we have protection at other layers of the system (especially, Cassandra actually won't start adn this code will never run if you try to add the `full_query_logging_options` into cassandra.yaml on 3x).

This is just here for completeness.